### PR TITLE
Fix gz-cmake declaration on package.xml (fix windows CI)

### LINKED ISSUE #1004: :woman_farmer: Error finding gz-cmake3 package build regression
## Environment
Gz version: Harmonic
Affected builds: 
- https://build.osrfoundation.org/job/gz_sim-8-win
- https://build.osrfoundation.org/job/gz_rendering-8-win/
- https://build.osrfoundation.org/job/gz_launch-7-win
- https://build.osrfoundation.org/job/gz_gui-8-win
## Description
* Expected behavior: Builds to pass and find gz-cmake3 
* Actual behavior: Builds fail finding gz-cmake3 even though that is built from source from the job. 

## Steps to reproduce

1. Run a build of any of the jobs listed above
2. See it fail
## Output
```
CMake Error at C:/vcpkg/scripts/buildsystems/vcpkg.cmake:782 (_find_package):
  By not providing "Findgz-cmake3.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "gz-cmake3", but CMake did not find one.

  Could not find a package configuration file provided by "gz-cmake3" with
  any of the following names:

    gz-cmake3Config.cmake
    gz-cmake3-config.cmake

  Add the installation prefix of "gz-cmake3" to CMAKE_PREFIX_PATH or set
  "gz-cmake3_DIR" to a directory containing one of the above files.  If
  "gz-cmake3" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:12 (find_package)
```

### PR DESCRIPTION
**Pull request by j-rivero**
# 🦟 Bug fix

Fix #1004 

## Summary
The build dependency is broken declaring gz-cmake instead of gz-cmake3. This is probably making colcon to ignore the gz-cmake3 install directory thus there is not gz-cmake3 config file. 

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

### CODE REVIEW

**Review by azeey — APPROVED** [review]
Thanks for the fix! I'm not sure why CI passed in the PR(#981) that added this though.

**Comment by j-rivero** [comment]
> Thanks for the fix! I'm not sure why CI passed in the PR(#981) that added this though.

weird indeed.

**Review by scpeters — APPROVED** [review]

**Comment by j-rivero** [comment]
can not merge it myself de to the lack of permissions and the failing tests :cry:

**Comment by azeey** [comment]
The failure seems unrelated, so I'll go ahead and merge to unblock other PRs.

**Pull request merged**
